src/terminal/scroll.lua

```diff
@@ -29,11 +33,25 @@ function M.scroll_regions(top, bottom)
 end
 
 --- Sets the scroll region and writes it to the terminal.
--- @tparam[opt] number top Top row of the scroll region.
--- @tparam[opt] number bottom Bottom row of the scroll region.
+-- Negative indices are supported, counting from the bottom of the screen.
+-- For example, `-1` refers to the last row, `-2` refers to the second-to-last row, etc.
+-- @tparam number start_row The first row of the scroll region (can be negative).
+-- @tparam number end_row The last row of the scroll region (can be negative).
 -- @return true
-function M.scroll_region(top, bottom)
-  output.write(M.scroll_regions(top, bottom))
+-- @within scroll_region
+function M.scroll_region(start_row, end_row)
+  -- Input validation
+  assert(type(start_row) == "number", "start_row must be a number")
+  assert(type(end_row) == "number", "end_row must be a number")
+
+  -- Resolve negative indices
+  start_row = resolve_index(start_row, get_height())
+  end_row = resolve_index(end_row, get_height())
+
+  -- Ensure start_row is less than or equal to end_row
+  assert(start_row <= end_row, "start_row must be less than or equal to end_row")
+
+  output.write("\27[" .. tostring(start_row) .. ";" .. tostring(end_row) .. "r")
```

Comment:
All of this logic should go in the `scroll_regions` function above, so from here we can just call that function. Similar as the other comment, with `cursor_set`.